pkg/segment/reader/segread/segstatsreader_test.go

```diff
@@ -190,3 +190,330 @@ func TestGetSegList(t *testing.T) {
 		})
 	}
 }
+
+func TestGetSegAvg_CurrSegStatIsNil(t *testing.T) {
+	runningSegStat := &structs.SegStats{}
+	currSegStat := (*structs.SegStats)(nil)
+
+	expected := &utils.NumTypeEnclosure{
+		Ntype:    utils.SS_DT_FLOAT,
+		IntgrVal: 0,
+	}
+	result, err := GetSegAvg(runningSegStat, currSegStat)
+
+	assert.Error(t, err)
+	assert.Equal(t, expected, result)
```

Comment:
nit: for cases where you expect an error, you don't also have to test `assert.Equal(t, expected, result)` (but since you already have it, you can keep it)